Assets/Scripts/Actors/ActorComponents/ActorPhysics.cs

```diff
@@ -153,23 +156,32 @@ public void ClimbSurface()
 	{
 		if(climbSurface)
 		{
-			rigidbody.useGravity = false;
-			currStoppingPower = stoppingSpeed;
+			ClimbableTag ct = climbSurface.GetComponent<ClimbableTag>();
+			if(ct)
+			{
+				rigidbody.useGravity = false;
+				currStoppingPower = stoppingSpeed;
 
-			Vector3 adjInput = climbSurface.InverseTransformDirection( inputVec );
-			moveVec = transform.rotation * adjInput * climbMoveSpeed * moveSpeedMod;
+				Vector3 adjInput = inputVec;
+				adjInput = climbSurface.InverseTransformDirection( adjInput );
+				adjInput = new Vector3( adjInput.x, adjInput.z, adjInput.y );
 
-			lastVelocity = moveVec;
-			rigidbody.velocity = moveVec;
+				if ( !ct.XMovement ) adjInput.x = 0f;
+				if ( !ct.YMovement ) adjInput.y = 0f;
 
-			if ( actor.animator != null )
-			{
-				actor.animator.SetBool( "isMoving", true );
-			}
+				moveVec = climbSurface.rotation * adjInput * climbMoveSpeed * moveSpeedMod + climbSurface.forward * 0.1f;
```

Comment:
what's with the .1f here? Also consider breaking this line up into two if it doesn't all fit on one screen